terraform/dotnet/ec2/asg/main.tf

```diff
@@ -0,0 +1,263 @@
+  }
+}
+
+resource "null_resource" "remote_service_setup" {
+  connection {
+    type        = "ssh"
+    user        = var.user
+    private_key = local.private_key_content
+    host        = aws_instance.remote_service_instance.public_ip
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      <<-EOF
+      #!/bin/bash
+
+      # Install DotNet and wget
+      sudo yum install wget -y
+      sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+      sudo wget -O /etc/yum.repos.d/microsoft-prod.repo https://packages.microsoft.com/config/fedora/37/prod.repo
+      sudo dnf install -y dotnet-sdk-8.0
+      sudo yum install unzip -y
+
+      # Copy in CW Agent configuration
+      agent_config='${replace(replace(file("./amazon-cloudwatch-agent.json"), "/\\s+/", ""), "$REGION", var.aws_region)}'
+      echo $agent_config > amazon-cloudwatch-agent.json
+
+      # Get and run CW agent rpm
+      ${var.get_cw_agent_rpm_command}
+      sudo rpm -U ./cw-agent.rpm
+      sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:./amazon-cloudwatch-agent.json
+
+      # Get ADOT distro and unzip it
+      ${var.get_adot_distro_command}
+
+      # Get and run the sample application with configuration
+      aws s3 cp ${var.sample_app_zip} ./dotnet-sample-app.zip
+      unzip -o dotnet-sample-app.zip
+
+      # Get Absolute Path
+      current_dir=$(pwd)
+      echo $current_dir
+
+      # Export environment variables for instrumentation
+      cd ./asp_remote_service
+      export CORECLR_ENABLE_PROFILING=1
+      export CORECLR_PROFILER={918728DD-259F-4A6A-AC2B-B85E1B658318}
+      export CORECLR_PROFILER_PATH=$current_dir/dotnet-distro/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so
+      export DOTNET_ADDITIONAL_DEPS=$current_dir/dotnet-distro/AdditionalDeps
+      export DOTNET_SHARED_STORE=$current_dir/dotnet-distro/store
+      export DOTNET_STARTUP_HOOKS=$current_dir/dotnet-distro/net/OpenTelemetry.AutoInstrumentation.StartupHook.dll
+      export OTEL_DOTNET_AUTO_HOME=$current_dir/dotnet-distro
+      export OTEL_DOTNET_AUTO_PLUGINS="AWS.Distro.OpenTelemetry.AutoInstrumentation.Plugin, AWS.Distro.OpenTelemetry.AutoInstrumentation"
+      export OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf
+      export OTEL_EXPORTER_OTLP_ENDPOINT=http://127.0.0.1:4316
+      export OTEL_AWS_APPLICATION_SIGNALS_EXPORTER_ENDPOINT=http://127.0.0.1:4316/v1/metrics
+      export OTEL_METRICS_EXPORTER=none
+      export OTEL_RESOURCE_ATTRIBUTES=service.name=dotnet-sample-remote-application-${var.test_id}
+      export OTEL_AWS_APPLICATION_SIGNALS_ENABLED=true
+      export OTEL_TRACES_SAMPLER=always_on
+      export ASPNETCORE_URLS=http://0.0.0.0:8081
+      dotnet build
+      nohup dotnet bin/Debug/netcoreapp8.0/asp_remote_service.dll &
+
+      # The application needs time to come up and reach a steady state, this should not take longer than 30 seconds
+      sleep 30
+
+      EOF
+    ]
```

Comment:
You need to check if the endpoint came up

Reply:
I believe we cannot do that since long time ago (private endpoint for sample app)

Reply:
You can check localhost:8081 in this case. Check the terraform for other languages, there are examples there.

Reply:
Added